src/domain/accounts/index.types.d.ts

```diff
@@ -136,13 +137,33 @@ type AccountValidator = {
   validateWalletForAccount(wallet: Wallet): true | ValidationError
 }
 
+type NewAccountWithEmailIdentifier = {
+  kratosUserId: KratosUserId
+  phone?: undefined
+  phoneMetadata?: undefined
```

Comment:
good catch. I've been struggling a little bit with the type here. 

it's because I have a `NewAccount` type below, and typescript would not be happy if there was no `phone` or `phoneMetadata` properties

Reply:
This ties into my previous comment that `NewAccount` is probably 2 types. `NewAccountWithPhoneIdenitty`, `NewAccountWithEmailIdentity` or something like that.

Reply:
In general `optional?` fields are almost always indications / symptoms of different use-cases being unified. Should be avoided / questioned when we encounter them.

Reply:
probably is better to use `Omit` https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys to create the new type

Reply:
thanks for your input @dolcalmi. this will go away in the next PR so not spending much time on this right now.